resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -104,6 +104,20 @@ private[spark] object Config extends Logging {
       .stringConf
       .createOptional
 
+  val KUBERNETES_EXECUTOR_LIMIT_GPUS =
```

Comment:
Why not allow the driver to request GPU's as well?

Reply:
Would drivers need GPU acceleration? My assumption was the executor code was where all the possibly acceleration would be needed.

Reply:
Sometimes you need it. For example, to reduce data across multiple executors, you would ideally use ring all-reduce among your executors, but you cannot really do that right now given that executors are scheduled independently. The best you can do right now is to gather all of your data to the driver and then do the reduction there. You can learn more at the SPIP for project hydrogen/barrier execution.

Reply:
If this needs to be shared it should go in a separate feature step that is applied in both the driver builder and the executor builder. There's probably a better way to share the resource requests / limits code here, probably by extracting the shared parts into a separate step that is parameterized by different values from the Spark configuration.